undertow/src/main/java/org/wildfly/swarm/undertow/runtime/HTTPSCustomizer.java

```diff
@@ -44,38 +48,58 @@
 
-    public HTTPSCustomizer() {
-    }
+    @Inject
+    @ConfigurationValue(SwarmProperties.HTTPS_GENERATE_SELF_SIGNED_CERTIFICATE)
+    private Boolean generateSelfCertificate;
 
     public void customize() {
         if (!this.managementCoreService.isUnsatisfied()) {
             UndertowFraction fraction = undertowInstance.get();
-            if (fraction.keystorePassword() != null & fraction.keystorePassword() != null && fraction.alias() != null) {
+            if ((generateSelfCertificate != null && generateSelfCertificate) || (fraction.keystorePassword() != null & fraction.keystorePassword() != null && fraction.alias() != null)) {
                 ManagementCoreService management = this.managementCoreService.get();
                 if (management == null) {
-                    throw new RuntimeException("HTTPS configured but org.wildfly.swarm:management not available");
+                    throw SwarmMessages.MESSAGES.httpsRequiresManagementFraction();
                 }
 
-                List<Server> servers = fraction.subresources().servers();
-
-                for (Server server : servers) {
+                for (Server server : fraction.subresources().servers()) {
                     if (server.subresources().httpsListeners().isEmpty()) {
                         server.httpsListener("default-https", (listener) -> {
-                            listener.securityRealm("SSLRealm");
-                            listener.socketBinding("https");
+                            listener.securityRealm("SSLRealm")
+                                    .socketBinding("https");
                         });
                     }
                 }
-
-                management.securityRealm("SSLRealm", (realm) -> {
-                    realm.sslServerIdentity((identity) -> {
-                        identity.keystorePath(fraction.keystorePath());
-                        identity.keystorePassword(fraction.keystorePassword());
-                        identity.alias(fraction.alias());
+                if (generateSelfCertificate != null && generateSelfCertificate) {
+                    // Remove when SWARM-634 is fixed
+                    if (System.getProperty("jboss.server.data.dir") == null) {
+                        File tmpDir = null;
+                        try {
+                            tmpDir = TempFileManager.INSTANCE.newTempDirectory("wildfly-swarm-data", ".d");
+                            System.setProperty("jboss.server.data.dir", tmpDir.getAbsolutePath());
+                        } catch (IOException e) {
+                            // Ignore
+                        }
+                    }
+                    management.securityRealm("SSLRealm", (realm) -> {
+                        realm.sslServerIdentity((identity) -> {
+                            identity.keystorePath("application.keystore")
+                                    .keystorePassword("password")
```

Comment:
I was assuming that keystorePath() and password wouldn't be provided in this case